### ISSUE TITLE
Flutter demo app: Could not find com.android.tools.build:gradle:7.6.3.

### ISSUE THREAD

**Issue by mihalycsaba** [flutter/flutter#148996]
### Steps to reproduce

1. flutter create test
2. flutter build apk

### Actual results

```
$ flutter build apk
Resolving dependencies... 
Downloading packages... 
  flutter_lints 3.0.2 (4.0.0 available)
  leak_tracker 10.0.4 (10.0.5 available)
  leak_tracker_flutter_testing 3.0.3 (3.0.5 available)
  lints 3.0.0 (4.0.0 available)
  material_color_utilities 0.8.0 (0.11.1 available)
  meta 1.12.0 (1.15.0 available)
  test_api 0.7.0 (0.7.1 available)
  vm_service 14.2.1 (14.2.2 available)
Got dependencies!
8 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':gradle:compileGroovy'.
> Could not resolve all files for configuration ':gradle:compileClasspath'.
   > Could not find com.android.tools.build:gradle:7.6.3.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/7.6.3/gradle-7.6.3.pom
       - https://repo.maven.apache.org/maven2/com/android/tools/build/gradle/7.6.3/gradle-7.6.3.pom
     Required by:
         project :gradle

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
Running Gradle task 'assembleRelease'...                         1,594ms
Gradle task assembleRelease failed with exit code 1
```

### Logs

<details open>
<summary>Logs</summary>

```console

<!-- Paste your logs here -->

```

</details>

### Flutter Doctor output

<details open>
<summary>Doctor output</summary>

```console

$ flutter doctor -v
[✓] Flutter (Channel stable, 3.22.1, on EndeavourOS 6.9.1-zen1-1.1-zen, locale en_US.UTF-8)
    • Flutter version 3.22.1 on channel stable at /nvme-data/Android/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a14f74ff3a (28 hours ago), 2024-05-22 11:08:21 -0500
    • Engine revision 55eae6864b
    • Dart version 3.4.1
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /ssd2/csaba/Android/Sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /ssd2/csaba/Android/Sdk
    • Java binary at: /opt/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11572160)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 17.0.6
    • cmake version 3.29.3
    • ninja version 1.12.1
    • pkg-config version 2.1.1

[✓] Android Studio (version 2023.3)
    • Android Studio at /opt/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11572160)

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • EndeavourOS 6.9.1-zen1-1.1-zen
    • Chrome (web)    • chrome • web-javascript • Google Chrome 125.0.6422.76

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by huycozy** [comment]
Hi @mihalycsaba, there is a similar issue at https://github.com/flutter/flutter/issues/130667. Please look a https://github.com/flutter/flutter/issues/130667#issuecomment-1637541437, it's pointing out that the AGP version you are using doesn't exist on https://mvnrepository.com/artifact/com.android.tools.build/gradle?repo=google. You should use one of available versions there. 

Closing this as it's more like a Gradle issue than a Flutter framework issue.

**Comment by mihalycsaba** [comment]
@huycozy I have tried a lot of things for hours. setting different Gradle and AGP versions, deleting every possible cache and Android Studio settings. I don't understand gradle enough. 

It is just weird that the default Demo app fails. I just need help with troubleshooting.

**Comment by huycozy** [comment]
Could you please share the entire output of `flutter build apk -v`? Thanks!

**Comment by mihalycsaba** [comment]
@huycozy It's weird it works now. I have noticed it still uses 7.6.3, saw this in the output `/.gradle/daemon/7.6.3/daemon-73102.out.log ` Maybe something was up with the servers a few days ago and now it managed to download it.

It was weird I couldn't change the version to 7.6.4 the other day changing the `distributionUrl`, I have tried many times and it only wanted to download 7.6.3 didn't matter what version I tried, I have tried a few. But now it works `.gradle/daemon/7.6.4/daemon-77868.out.log `, I can change it back too to the older version...
